lua/rustaceanvim/rust_analyzer.lua

```diff
@@ -5,18 +5,82 @@ local os = require('rustaceanvim.os')
 ---@class rustaceanvim.rust-analyzer.ClientAdapter
 local M = {}
 
+--- Default target value for rustc when no specific target is provided.
+--- Used as a fallback to let rustc determine the appropriate target based on the OS.
+DEFAULT_RUSTC_TARGET = 'OS'
+
+---Local rustc targets cache
+local rustc_targets_cache = nil
+
+M.load_os_rustc_target = function()
+  vim.system({ 'rustc', '-Vv' }, { text = true }, function(result)
+    if result.code == 0 then
+      for line in result.stdout:gmatch('[^\r\n]+') do
+        local host = line:match('^host:%s*(.+)$')
+        if host then
+          M.os_rustc_target = host
+          break
+        end
+      end
+    end
+  end)
+end
+
+---Handles retrieving rustc target architectures and running the passed in callback
+---to perform certain actions using the retrieved targets.
+---@param callback fun(targets: string[])
+M.with_rustc_target_architectures = function(callback)
```

Comment:
issue: this isn't actually used. There's a duplicate in the `lsp` module that is used.
I think the `lsp` module (local function) or a `rustaceanvim.rustc` module is a better place for it, because this function doesn't actually interact with rust-analyzer.

Reply:
agreed!